File: medium/3634-minimum-removals-to-balance-array/solution.java
import java.util.Arrays;

/**
 * 3634. Minimum Removals to Balance Array
 * Time: O(n log n)
 * Space: O(1)
 */
class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);
        int n = nums.length;
        
        int l = 0;
        int maxLen = 1;
        
        for (int r = 0; r < n; r++) {
            while ((long) nums[r] > (long) nums[l] * k) {
                l++;
            }
            maxLen = Math.max(maxLen, r - l + 1);
        }
        
        return n - maxLen;
    }
}

File: medium/3634-minimum-removals-to-balance-array/solution.py
from typing import List


class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        """
        3634. Minimum Removals to Balance Array
        Time: O(n log n)
        Space: O(1)
        """
        nums.sort()
        n = len(nums)
        
        l = 0
        max_len = 1
        
        for r in range(n):
            while nums[r] > nums[l] * k:
                l += 1
            max_len = max(max_len, r - l + 1)
        
        return n - max_len
